hbase-server/src/main/java/org/apache/hadoop/hbase/ipc/RpcServer.java

```diff
@@ -324,6 +330,13 @@ public void onConfigurationChange(Configuration newConf) {
       refreshAuthManager(newConf, new HBasePolicyProvider());
     }
     refreshSlowLogConfiguration(newConf);
+    if (
+      CoprocessorConfigurationUtil.checkConfigurationChange(getConf(), newConf,
+        CoprocessorHost.RPC_COPROCESSOR_CONF_KEY)
+    ) {
+      LOG.info("Update the RPC coprocessor(s) because the configuration has changed");
+      initializeCoprocessorHost(newConf);
```

Comment:
Is a Configuration object enough for your usage? I see the RpcCoprossesorEnvironment is also empty?

Reply:
Strictly for this particular hook we don't anything else. Configuration object is needed to instantiate Observers, the rest is passed in method arguments.